File: 1518-water-bottles/1518-water-bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int drinkedBottles = numBottles;

        while(numBottles >= numExchange) {
            int exchangeableBottles = numBottles / numExchange;
            int leftEmptyBottles = numBottles % numExchange;

            drinkedBottles += exchangeableBottles;
            numBottles = exchangeableBottles + leftEmptyBottles;
        }

        return drinkedBottles;
    }
}

File: 1518-water-bottles/1518-water-bottles.c
int numWaterBottles(int numBottles, int numExchange) {

    int drinkedBottles = numBottles;

    while (numBottles >= numExchange) {

        int exchangeBottles = numBottles / numExchange;
        drinkedBottles += exchangeBottles;
        int emptyBottles = numBottles % numExchange;

        numBottles = exchangeBottles + emptyBottles;
    }

    return drinkedBottles;
}